src/lrp.py

```diff
@@ -16,58 +18,58 @@
 
 
 def remove_pad(pt: bytes):
-    padl = 0
+    padding = 0
 
     for b in pt[::-1]:
-        padl += 1
+        padding += 1
 
         if b == 0x80:
             break
 
         if b != 0x00:
             raise RuntimeError('Invalid padding')
 
-    return pt[:-padl]
+    return pt[:-padding]
 
 
-def nibbles(x: Union[bytes, str]) -> Generator[int, None, None]:
+def nibbles(val: Union[bytes, str]) -> Generator[int, None, None]:
     """
     Generate integers out of x (bytes), applicable for m = 4
     """
-    if isinstance(x, bytes):
-        x = x.hex()
+    if isinstance(val, bytes):
+        val = val.hex()
 
-    for nb in x:
+    for nb in val:
         yield binascii.unhexlify("0" + nb)[0]
 
 
-def incr_counter(r: bytes):
-    max_bit_len = len(r) * 8
+def incr_counter(record: bytes):
+    max_bit_len = len(record) * 8
```

Comment:
I think that it might be really misleading to rename variables here. Maybe the variable names are not perfect, but they match the NXP's application notes on LRP implementation.

In the quoted fragment, `r` is meaningful - this value appears in the algorithm's specification and I also wouldn't translate this as a `record`. That's just mathematical `r`.

Reply:
Thanks for the comments. I agree that consistency with NXP's notes is best to be preserved. I have reverted the renamed variables.